src/app/src/pages/redteam/setup/components/StrategyConfigDialog.tsx

```diff
@@ -70,7 +164,22 @@ export default function StrategyConfigDialog({
     setGoals(
       strategy === 'simba' ? nextConfig.goals || DEFAULT_SIMBA_GOALS : nextConfig.goals || [],
     );
-  }, [open, strategy, config]);
+    // Normalize steps: convert objects with {id, config} to just strings
+    const rawSteps = strategy === 'layer' ? nextConfig.steps || [] : [];
+    const normalizedSteps = rawSteps.map((step: any) =>
+      typeof step === 'string' ? step : step.id,
+    );
+    setSteps(normalizedSteps);
+    // Filter layerPlugins to only include plugins that are in availablePlugins
+    const configPlugins = strategy === 'layer' ? nextConfig.plugins || [] : [];
```

Comment:
I really don't like the `strategy === 'layer'` bits of this code- this will be so annoying to change if we ever add another strategy that needs to interact here. Can we do a singular `if (strategy === 'layer')` conditional and apply only that logic for layer, rather than interspersing it?